src/js/client.js

```diff
@@ -184,143 +182,144 @@ function Client(username, password) {
-      return
-    }
+  packetHandler_0x68_pingB: function(packet) {
+    var timestamp = packet.readInt32();
 
-    var x62 = new ClientPacket(0x62)
-    x62.writeByte(0x34)
-    x62.writeByte(0x00)
-    x62.writeByte(0x0A)
-    x62.writeByte(0x88)
-    x62.writeByte(0x6E)
-    x62.writeByte(0x59)
-    x62.writeByte(0x59)
-    x62.writeByte(0x75)
-    _client.send(x62)
-
-    var x00 = new ClientPacket(0x00)
-    x00.writeInt16(_client.daVersion)
-    x00.writeByte(0x4C)
-    x00.writeByte(0x4B)
-    x00.writeByte(0x00)
-    _client.send(x00)
-
-    _client.sentVersion = true
-  }
+    var x75 = new ClientPacket(0x75);
+    x75.writeInt32(timestamp);
+    x75.writeInt32(int32(this.tickCount()));
+    this.send(x75);
+  },
 
-  this.handleRecv = function(recvBuffer) {
-    recvBuffer = new Uint8Array(recvBuffer)
+  packetHandler_0x7E_welome: function(packet) {
+    if (this.sentVersion) {
+      return;
+    }
 
-    if (recvBuffer.length == 0) {
-      _client.disconnect()
-      return
+    var x62 = new ClientPacket(0x62);
+    x62.writeByte(0x34);
+    x62.writeByte(0x00);
+    x62.writeByte(0x0A);
+    x62.writeByte(0x88);
+    x62.writeByte(0x6E);
+    x62.writeByte(0x59);
+    x62.writeByte(0x59);
+    x62.writeByte(0x75);
+    this.send(x62);
+
+    var x00 = new ClientPacket(0x00);
+    x00.writeInt16(_client.daVersion);
+    x00.writeByte(0x4C);
+    x00.writeByte(0x4B);
+    x00.writeByte(0x00);
+    this.send(x00);
+
+    this.sentVersion = true;
+  },
+
+  receive: function(buffer) {
+    if (!buffer) {
+      this.disconnect();
+      return;
     }
 
-    while (recvBuffer.length > 3) {
-      if (recvBuffer[0] != 0xAA) {
-        return
+    while (buffer.length > 3) {
+      if (buffer[0] != 0xAA) {
+        return;
       }
 
-      var length = recvBuffer[1] << 8 | recvBuffer[2] + 3
+      var length = buffer[1] << 8 | buffer[2] + 3;
 
-      if (length > recvBuffer.length) {
-        break
+      if (length > buffer.length) {
+        break;
       }
 
-      var buffer = recvBuffer.slice(0, length)
-      recvBuffer = recvBuffer.slice(length)
-
-      var packet = new ServerPacket(buffer)
+      var packetBuffer = Array.from(buffer.slice(0, length));
+      var packet = new ServerPacket(packetBuffer);
+      buffer = buffer.slice(length);
 
       if (packet.shouldEncrypt()) {
-        packet.decrypt(_client.crypto)
+        packet.decrypt(this.crypto);
       }
 
-      if (_client.showIncoming) {
-        console.log("Received: {0}".format(packet.toString()))
+      if (this.logIncoming) {
+        console.log("Received: {0}".format(packet.toString()));
       }
 
-      if (packet.opcode in _client.packetHandlers) {
-          _client.packetHandlers[packet.opcode](packet)
+      if (packet.opcode in this.packetHandlers) {
+          this.packetHandlers[packet.opcode](packet);
       }
     }
-  }
+  },
 
-  this.packetHandlers = {
+  packetHandlers: {
     0x00: this.packetHandler_0x00_encryption,
```

Comment:
These functions appear to be undefined now:
`var c = new Client("pyda", "pdya")`
undefined
`c.packetHandlers`
Object {0: undefined, 2: undefined, 3: undefined, 5: undefined, 10: undefined, 13: undefined, 59: undefined, 76: undefined, 104: undefined, 126: undefined}